packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -402,24 +412,53 @@ function buildTree(rootStack, readHookLog): HooksTree {
       subHooks: [],
     });
   }
+
+  // Associate custom hook values (useInpect() hook entries) with the correct hooks
+  rootChildren.forEach(hooksNode => rollupDebugValues(hooksNode));
+
   return rootChildren;
 }
 
+function rollupDebugValues(hooksNode: HooksNode): void {
+  let useInpectHooksNodes: Array<HooksNode> = [];
+  hooksNode.subHooks = hooksNode.subHooks.filter(subHooksNode => {
```

Comment:
Maybe just make this into a for-loop. Like Dan, I find it confusing if something is both functional style and imperative. I assume this to be pure if it's using `filter`.